packages/flutter/lib/src/painting/gradient.dart

```diff
@@ -240,6 +240,12 @@ abstract class Gradient {
   /// Typically this is the same as interpolating from null (with [lerp]).
   Gradient scale(double factor);
 
+  /// Returns a new gradient that matches this gradient with the given opacity
+  ///
+  /// A factor of 0.0 (or less) should result in a variant of the gradient that
+  /// is fully transparent
```

Comment:
```suggestion
  /// Returns a new [Gradient] with each color set to the given opacity.
```

(I think the word "should" has an implication of uncertainty, and I personally don't believe that someone not understanding "zero opacity" will be an issue.)

Reply:
Since the other methods `@override` this one, doc comments aren't necessary for them. Feel free to delete each instance of:
```dart
/// Returns a new [LinearGradient] with its colors opacity changed to the given opacity.
```
(Currently, the grammar is not ideal, and each one says `[LinearGradient]` instead of matching the class name.)